staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -175,8 +189,54 @@ func (c *AvailableConditionController) sync(key string) error {
 			return err
 		}
 	}
+	// actually try to hit the discovery endpoint when it isn't local and when we're routing as a service.
+	if apiService.Spec.Service != nil && c.serviceResolver != nil {
+		discoveryURL, err := c.serviceResolver.ResolveEndpoint(apiService.Spec.Service.Namespace, apiService.Spec.Service.Name)
+		if err != nil {
+			return err
+		}
+		// construct an http client that will ignore TLS verification (if someone owns the network and messes with your status
+		// that's not so bad) and sets a very short timeout.
+		httpClient := &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			},
+			// the request should happen quickly.
+			Timeout: 5 * time.Second,
+		}
+		if c.proxyTransport != nil {
+			httpClient.Transport = c.proxyTransport
+		}
+
+		errCh := make(chan error)
+		go func() {
+			resp, err := httpClient.Get(discoveryURL.String())
+			if resp != nil {
+				resp.Body.Close()
+			}
+			errCh <- err
+		}()
 
-	// TODO actually try to hit the discovery endpoint
+		select {
+		case err = <-errCh:
+		case <-time.After(6 * time.Second):
```

Comment:
make it a constant. Why 6? At least it's a [prefect number](https://en.wikipedia.org/wiki/Perfect_number).

Reply:
And you don't trust the client's timeout. Why?

Reply:
> And you don't trust the client's timeout. Why?

We had trouble with it downstream when the dialing or dns lookups were failing.

Reply:
add a comment to that nobody clean that up and removes the second timeout.

Reply:
> add a comment to that nobody clean that up and removes the second timeout.

done